src/Task/MapRoute.pm

```diff
@@ -271,8 +273,66 @@ sub iterate {
 
 	} elsif ( $portals_lut{"$self->{mapSolution}[0]{map} $self->{mapSolution}[0]{pos}{x} $self->{mapSolution}[0]{pos}{y}"}{source} ) {
 		# This is a portal solution
-
-		if ( distance($self->{actor}{pos_to}, $self->{mapSolution}[0]{pos}) < 2 ) {
+		
+		if ($self->{missing_portal}) {
+		
+			if (!$config{route_tryToGuessMissingPortalByDistance} || $portalsList->size == 0) {
```

Comment:
This doesn't obey `guess_skip`. Maybe `$portalsList->size <= keys %{$self->{guess_skip} || {}}`?